Automattic/wp-calypso
# Enable localization of read/search page with url param

Adds support for localization to the logged out read/search page so that it can be a suitable replacement for the public search page.

The approach recomemnded in pe7F0s-ZH-p2#comment-1126 was to use the same scheme as existing public pages such as `/themes` which includes the language code in the URL e.g. `/fr/themes`.

I was not able to use the exact same approach as those pages because they use the `ssrSetupLocale` middleware which only works for server side rendered pages, however the search page does not currently support server side rendering.

This approach works on the client however using the existing `setLocaleMiddleware`. The `lang` param is also passed along to the search endpoint. This will also have the affect of localizing other reader stream endpoints. I also updated the search suggestions to be localized if the user is logged out.


### Testing instructions

In an incognito window go to a localized route such as /fr/read/search
Search results should also be localized.
